src/Common/src/System/Diagnostics/TraceListenerHelpers.Windows.cs

```diff
@@ -0,0 +1,36 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+namespace System.Diagnostics
+{
+    internal static partial class TraceListenerHelpers
+    {
+        private static volatile int s_processId;
+
+        internal static int GetProcessId()
+        {
+            EnsureProcessInfo();
+            return s_processId;
+        }
+
+        internal static string GetProcessName()
+        {
+            EnsureProcessInfo();
+            return s_processName;
+        }
+
+        private static void EnsureProcessInfo()
+        {
+            if (s_processName == null)
+            {
+                using (var process = Process.GetCurrentProcess())
+                {
+                    s_processName = process.ProcessName;
+                    s_processId = process.Id;
```

Comment:
Maybe we don't care, but this isn't thread safe.  Imagine two threads racing to access GetProcessName and GetProcessId.  The first thread calls GetProcessName, sees that s_processName is null, gets the name and stores it into s_processName.  At that moment, the other thread calls GetProcessId, sees that s_processName is not null, and returns s_processId, which is still 0.

Reply:
Since both s_processName and s_processId are volatile, the issue should go away if you swapped the order of these two lines.  That way, if any thread read s_processName != null, it would also see s_processId correctly.

Reply:
Right, good catch 😄